### DISCUSSION TITLE
I don't know what initialReduxState is

### DISCUSSION THREAD

**Discussion by hide1111111** [vercel/next.js#65773] [Help] [answered]
### Summary

I have this error
Error: Cannot read properties of undefined (reading 'initialReduxState')
please help me
I couldn't figure it out with other people's solutions

### Additional information

```js
layout.js

import { Provider } from 'react-redux'
import { useStore } from '../store'
import Layout from '../components/Layout'
import '../globals.css'

function MyApp({ Component, pageProps }) {
  const store = useStore(pageProps.initialReduxState)

  return (
    <Provider store={store}>
      <Layout>
        <Component {...pageProps} />
      </Layout>
    </Provider>
  )
}

export default MyApp


store.js

export function useStore(initialState) {
  const store = useMemo(() => initializeStore(initialState), [initialState])
  return store
}
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

You are mixing up concepts here. `{Component, pageProps}`, where some of the props passed to the custom `App` component, that lived in _app, in Pages Router.

You are however, showing a `layout.js` file, which indicates that you are in the App Router.

Are you following a tutorial or something like that? If you used create-next-app, you might have accidentally chose to use App Router, over Pages Router, when it asked you which one you'd want to use.

**Reply by hide1111111** [reply]
thank you
I was able to understand
